Better test descriptions

nrichers opened the pull request:
## Internal Notes for Reviewers

This PR embeds the files from @cachafla's https://github.com/validmind/developer-framework/pull/282 in a new landing page. It's basically the `tests/` folder copied over into `site/` and then the files get embedded a couple of times:

- In the left sidebar via `_quarto.yml`
- In the new landing page in `guide/test-descriptions.qmd`

No Makefile updates, yet, as this is just a demo.

Note we have the option to expand the left sidebar but, if you do, it makes for a loooong list of tests. Just using the landing page with the tests collapsed to the level shown in the next section might be preferrable. We can also specify how many tests are shown for each section. (The tool I used for the screenshot doesn't quite show the sidebar accurately, FYI.)

<img width="199" alt="image" src="https://github.com/validmind/documentation/assets/15148011/42f3dc02-eb84-497a-8866-34534ea28d01">

## Output

![Capto_Capture 2023-11-02_11-25-57_am](https://github.com/validmind/documentation/assets/15148011/1a652986-4d14-46b9-b823-7fbf479d23e1)

<!--
PR instructions for release notes:

1. Pick at least one label:

- `internal` (skip Step 2, no release notes required)
- `highlight`
- `enhancement`
- `bug`
- `deprecation`
- `documentation`

2. In the next section, describe the changes so that an external user can understand them. Keep it simple and link to the docs with [Learn more ...](<relative-link>), if available.
-->

## External Release Notes

We added better descriptions for the tests that are available in the ValidMind Developer Framework along with a new landing page where you can browse through all of the tests. Our docs site search now also indexes these descriptions, making them easier to find. [Try it ...](../../guide/test-descriptions.qmd)

cachafla reviewed (approved):
Looks great 👍 

I left a related comment here: https://github.com/validmind/developer-framework/pull/282#issuecomment-1791544352

Let's go with this v1 and we can improve upon the format and metadata of the tests later.

nrichers commented:
Updated the Makefile to make the test descriptions part of the source we fetch from the developer-framework repo and added a release notes description with a link. Also switched to a tabbed view after chatting with Andres some more. 

![2023-11-02_18-29-08 (1)](https://github.com/validmind/documentation/assets/15148011/83117eff-d8b3-49a9-ac66-ef32852cee6b)
